src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -39,15 +38,15 @@ namespace Microsoft.CodeAnalysis
         /// The path which contains mscorlib.  This can be null when specified by the user or running in a 
         /// CoreClr environment.
         /// </summary>
-        internal string SdkDirectory { get; }
+        internal string? SdkDirectory { get; }
 
         /// <summary>
         /// The temporary directory a compilation should use instead of <see cref="Path.GetTempPath"/>.  The latter
         /// relies on global state individual compilations should ignore.
         /// </summary>
-        internal string TempDirectory { get; }
+        internal string? TempDirectory { get; }
```

Comment:
Considered calling this `CustomTempDirectory` as it's really used when we override `Path.GetTempPath`. I think it makes the intent clearer. Open to doing this but wanted to hear feedback from others.

Reply:
Or maybe `ExplicitTempDirectory`

Reply:
Eh. I think it's fine.